File: practica1/src/gestorAplicacion/personas/Persona.java
package gestorAplicacion.personas;

import java.io.Serializable;
import java.time.*;
import java.util.ArrayList;


public abstract class Persona implements Serializable{

	//ATRIBUTOS DE CLASE
	private static final long serialVersionUID = 1L;
	protected static ArrayList<Persona> lista = new ArrayList<>();

	
	//ATRIBUTOS INSTANCIA
	protected String nombre;
	protected int id;
	protected String correo;
	protected int tel;
	protected String direccion;
	protected LocalDate nacimiento;
	protected String paisOrigen;
	protected boolean vivo=true;
	
	//CONSTRUCTORES
	Persona(){};
	Persona (int id,String nombre, LocalDate nacimiento, String paisOrigen){
		this.id=id;
		this.nombre=nombre;
		this.nacimiento=nacimiento;
		this.paisOrigen=paisOrigen;
	}
	Persona(String nombre,int id,String correo,int tel,String direccion,LocalDate nacimiento, String paisOrigen){
		this(id,nombre, nacimiento, paisOrigen);
		this.correo=correo;
		this.tel=tel;
		this.direccion=direccion;
		this.paisOrigen=paisOrigen;
	}
	
	//METODOS
	public static void eliminarPersona(Persona p) { // elimina un registro de Persona
		lista.remove(lista.indexOf(p));
	}
	public static String mostrarRegistros() {
		String c="Personas creadas: "+"\n";
		for (int i = 0; i < lista.size(); i++) {
			if(lista.get(i) instanceof Autor ) {c = c + i+"."+"Autor "+ lista.get(i).nombre + "  ID "+ lista.get(i).id+ "\n";	}
			else if(lista.get(i) instanceof EstudianteProfesor) {c = c + i+"."+"Usuario de la Universidad "+ lista.get(i).nombre + "  ID (CP) "+ lista.get(i).getId()+ "\n";}
			else if(lista.get(i) instanceof Externo) {c = c + i+"."+"Usuario externo de la Universidad "+ lista.get(i).nombre + "  ID (CP)"+ lista.get(i).getId()+ "\n";}
		}
		return c;
	}
	
	//METODO ABSTRACTO
	public abstract String infoPersonal ();
		
		
	//GETTERS SETTERS
	
	public String getNombre() {
		return nombre;
	}
	public void setNombre(String nombre) {
		this.nombre = nombre;
	}
	public String getPaisOrigen() {
		return paisOrigen;
	}
	public void setPaisOrigen(String paisOrigen) {
		this.paisOrigen = paisOrigen;
	}
	public boolean isVivo() {
		return vivo;
	}
	public void setVivo(boolean vivo) {
		this.vivo = vivo;
	}
	public int getId() {
		return id;
	}
	public void setId(int id) {
		this.id = id;
	}
	public String getCorreo() {
		return correo;
	}
	public void setCorreo(String correo) {
		this.correo = correo;
	}
	public int getTel() {
		return tel;
	}
	public void setTel(int tel) {
		this.tel = tel;
	}
	public String getDireccion() {
		return direccion;
	}
	public void setDireccion(String direccion) {
		this.direccion = direccion;
	}
	public static ArrayList<Persona> getLista() {
		return lista;
	}
	public static void setLista(ArrayList<Persona> lista) {
		Persona.lista = lista;
	}
	public LocalDate getNacimiento() {
		return nacimiento;
	}
	public void setNacimiento(LocalDate nacimiento) {
		this.nacimiento = nacimiento;
	}
}

File: practica2-python/src/gestorAplicacion/personas/Persona.py
from abc import ABC, abstractmethod

#   CLASE ABSTRACTA
class Persona(ABC):

    #Atributo de Clase
    _lista = []

    #Constructor y atributos
    def __init__(self, nombre='...',id=0,correo='...',tel=0,direccion="...",nacimiento='...',pais='...',vivo=True):
        self.nombre = nombre
        self.id = id
        self.correo = correo
        self.telefono = tel
        self.direccion = direccion
        self.nacimiento = nacimiento
        self.pais = pais
        self.vivo = vivo
        Persona._lista.append(self)

    #Metodos
    @classmethod
    def eliminarPersona(cls,persona):
        from gestorAplicacion.personas.Autor import Autor
        from gestorAplicacion.personas.EstudianteProfesor import EstudianteProfesor
        from gestorAplicacion.personas.Externo import Externo   
        if isinstance(persona,Autor):
            Autor.getAutores().remove(persona)
        elif isinstance(persona,EstudianteProfesor):
            EstudianteProfesor.l.remove(persona)
        elif isinstance(persona,Externo):   
            Externo.l.remove(persona)
        Persona._lista.remove(persona)
        del persona

    

    @classmethod
    def mostrarRegistros(cls):
        from gestorAplicacion.personas.Autor import Autor
        from gestorAplicacion.personas.EstudianteProfesor import EstudianteProfesor
        from gestorAplicacion.personas.Externo import Externo
        
        c="PERSONAS CREADAS: \n"

        for i,x in enumerate(Persona._lista ):
            if isinstance(x,Autor):
                c=c + str(i) + f'. Autor {x.getNombre()} con Identificacion {x.getId()} \n'
            elif isinstance(x,EstudianteProfesor):
                c=c + str(i) + f'. Usuario de la Universidad {x.getNombre()} con Identificacion {x.getId()} \n'
            elif isinstance(x,Externo):
                c=c + str(i) + f'. Usuario externo {x.getNombre()} con Identifiacion {x.getId()} \n'
        return c

    @abstractmethod
    def infoPersonal(self):
        pass

    

    def getNombre(self):
        return self.nombre
    def setNombre(self,n):
        self.nombre = n

    def getId(self):
        return self.id
    def setId(self,i):
        self.id = i

    def getCorreo(self):
        return self.correo
    def setCorreo(self, c):
        self.correo = c

    def getTel(self):
        return self.telefono
    def setTel(self,t):
        self.telefono = t

    def getDireccion(self):
        return self.direccion
    def setDireccion(self,d):
        self.direccion = d
    
    def getNacimiento(self):
        return self.nacimiento
    def setNacimiento(self,n):
        self.nacimiento = n
    
    def getPais(self):
        return self.pais
    def setPais(self,p):
        self.pais = p

    def isVivo(self):
        return self.vivo
    def setVivo(self,v):
        self.vivo = v

    @classmethod
    def getLista(cls):
        return Persona._lista
    